pkg/activitypub/store/ariesstore/ariesstore.go

```diff
@@ -284,7 +284,7 @@ func (s *Provider) queryActivitiesByRef(refType spi.ReferenceType, query *spi.Cr
 	}
 
 	if len(refs) == 0 {
-		return &activityIterator{}, nil
+		return memstore.NewActivityIterator(nil, 0), nil
```

Comment:
It's a bit weird to use an iterator from another implementation. It may be better to just add a check for nil ariesIterator in each of the functions. Maybe something to look at in the future PR.

Reply:
Makes sense. Eventually I should be consolidating the two storage implementations, so it would make sense to do it then.

Reply:
^^ Just realized that last comment may have been a bit odd-sounding: By that I mean: I see what you're saying. I'll definitely do that in a future PR

Reply:
I also use the mem iterator in another function (although I actually pass in stuff to the struct)